api/APIValidation.ts

```diff
@@ -0,0 +1,84 @@
+import { ZodError, ZodTypeAny, z } from "zod";
+import { addLogHandler, consoleLogHandler, logger } from "../logging";
+import { APIMiddleware, TiFAPIInputContext, TiFAPIResponse } from "./TransportTypes";
+
+const log = logger("tif.api.validation")
+addLogHandler(consoleLogHandler())
+
+const zodValidation = async <T>(data: T, schema: ZodTypeAny): Promise<T | "failure"> => {
+  try {
+      return await schema.parseAsync(data);
+  } catch (e) {
+      if (e instanceof ZodError) {
+          log.error("Zod Schema Error Message", {
+              zodError: JSON.parse(e.message)
+          });
+      } else {
+          log.error(e);
+      }
+      return "failure";
+  }
+};
+
+export type ValidationResult = "invalid-request" | "unexpected-response" | "invalid-response" | "passed"
+
+export type ValidationResultParser = 
+  ((_: {validationStatus: Extract<ValidationResult, "invalid-request">, request: TiFAPIInputContext<any>} 
+    | {validationStatus: Extract<ValidationResult, "unexpected-response" | "invalid-response">, response: TiFAPIResponse<any>} 
+    | {validationStatus: Extract<ValidationResult, "passed">, response: TiFAPIResponse<any>}
+  ) => (TiFAPIResponse<any>))
+
+export const VALIDATE_REQUEST = 1 << 0
+export const VALIDATE_RESPONSE = 1 << 1
+export const VALIDATE_NONE = 0
+export type APICallValidateOptions = number
```

Comment:
```suggestion
export enum APICallValidationOptions {
  None = 0,
  Request = 1 << 0,
  Response = 1 << 1
}
```
Figured out that we can model this as an enum for type safety.